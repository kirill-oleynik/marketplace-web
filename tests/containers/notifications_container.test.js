const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const FavoritesContainer = require('../../app/containers/notifications_container').default;

const fakeStore = (state = {}) => ({
  subscribe() {},
  dispatch() {},
  getState() { return state; }
});

const event = 'Test';
const notifications = {
  event
};

const renderFavoritesContainer = () => (
  <FavoritesContainer
    t={(translation) => translation}
  />
);

describe('#render', () => {
  test('it renders correctly', () => {
    const store = fakeStore({
      notifications
    });

    const component = shallow(
      renderFavoritesContainer(),
      { context: { store } }
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#passedProps', () => {
  test('it passess props to component correctly', () => {
    const store = fakeStore({
      notifications
    });

    const component = shallow(
      renderFavoritesContainer(),
      { context: { store } }
    );

    const props = component.props();

    expect(props.event).toEqual(event);
    expect(props.hide).toBeInstanceOf(Function);
  });
});
