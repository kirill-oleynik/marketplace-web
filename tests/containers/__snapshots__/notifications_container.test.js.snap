// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<Translate(NotificationsContainer)
  event="Test"
  hide={[Function]}
  t={[Function]}
/>
`;
